packages/devtools_app/lib/src/screens/performance/panes/flutter_frames/flutter_frames_controller.dart

```diff
@@ -0,0 +1,207 @@
+// Copyright 2022 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+
+import 'package:collection/collection.dart';
+import 'package:flutter/foundation.dart';
+
+import '../../../../primitives/trace_event.dart';
+import '../../../../primitives/utils.dart';
+import '../../../../service/service_manager.dart';
+import '../../../../shared/globals.dart';
+import '../../performance_controller.dart';
+import '../../performance_model.dart';
+import '../../performance_screen.dart';
+import 'flutter_frame_model.dart';
+
+class FlutterFramesController extends PerformanceFeatureController {
+  FlutterFramesController(super.performanceController);
+
+  /// The currently selected timeline frame.
+  ValueListenable<FlutterFrame?> get selectedFrame => _selectedFrameNotifier;
+  final _selectedFrameNotifier = ValueNotifier<FlutterFrame?>(null);
+
+  /// The flutter frames in the chart.
+  ValueListenable<List<FlutterFrame>> get flutterFrames => _flutterFrames;
+  final _flutterFrames = ListValueNotifier<FlutterFrame>([]);
+
+  /// Whether we should show the Flutter frames chart.
+  ValueListenable<bool> get showFlutterFramesChart => _showFlutterFramesChart;
+  final _showFlutterFramesChart = ValueNotifier<bool>(true);
+  void toggleShowFlutterFrames(bool value) =>
+      _showFlutterFramesChart.value = value;
+
+  /// Whether flutter frames are currently being recorded.
+  ValueListenable<bool> get recordingFrames => _recordingFrames;
+  final _recordingFrames = ValueNotifier<bool>(true);
+
+  ValueListenable<bool> get badgeTabForJankyFrames => _badgeTabForJankyFrames;
```

Comment:
Nit: While we're here, maybe add doc comments for properties that don't have them?

Reply:
good idea. Will do this as a follow up.